include/boost/beast/core/flat_buffer.hpp

```diff
@@ -206,131 +245,159 @@ class basic_flat_buffer
     std::size_t
-    max_size() const
+    max_size() const noexcept
     {
         return max_;
     }
 
-    /// Return the maximum sum of input and output sizes that can be held without an allocation.
+    /// Return the maximum number of bytes, both readable and writable, that can be held without requiring an allocation.
     std::size_t
-    capacity() const
+    capacity() const noexcept
     {
         return dist(begin_, end_);
     }
 
-    /// Get a list of buffers that represent the input sequence.
+    /// Returns a constant buffer sequence representing the readable bytes
     const_buffers_type
-    data() const
+    data() const noexcept
     {
         return {in_, dist(in_, out_)};
     }
 
-    /** Get a list of buffers that represent the output sequence, with the given size.
+    /// Returns a constant buffer sequence representing the readable bytes
+    const_buffers_type
+    cdata() const noexcept
+    {
+        return data();
+    }
 
-        @throws std::length_error if `size() + n` exceeds `max_size()`.
+    /// Returns a mutable buffer sequence representing the readable bytes
+    mutable_data_type
+    data() noexcept
+    {
+        return {in_, dist(in_, out_)};
+    }
+
+    /** Returns a mutable buffer sequence representing writable bytes.
+    
+        Returns a mutable buffer sequence representing the writable
+        bytes containing exactly `n` bytes of storage. Memory may be
+        reallocated as needed.
+
+        All buffers sequences previously obtained using
+        @ref data or @ref prepare are invalidated.
 
-        @note All previous buffers sequences obtained from
-        calls to @ref data or @ref prepare are invalidated.
+        @param n The desired number of bytes in the returned buffer
+        sequence.
+
+        @throws std::length_error if `size() + n` exceeds `max_size()`.
     */
     mutable_buffers_type
     prepare(std::size_t n);
 
-    /** Move bytes from the output sequence to the input sequence.
+    /** Append writable bytes to the readable bytes.
+
+        Appends n bytes from the start of the writable bytes to the
+        end of the readable bytes. The remainder of the writable bytes
+        are discarded. If n is greater than the number of writable
+        bytes, all writable bytes are appended to the readable bytes.
 
-        @param n The number of bytes to move. If this is larger than
-        the number of bytes in the output sequences, then the entire
-        output sequences is moved.
+        All buffers sequences previously obtained using
+        @ref data or @ref prepare are invalidated.
 
-        @note All previous buffers sequences obtained from
-        calls to @ref data or @ref prepare are invalidated.
+        @param n The number of bytes to append. If this number
+        is greater than the number of writable bytes, all
+        writable bytes are appended.
     */
     void
-    commit(std::size_t n)
+    commit(std::size_t n) noexcept
```

Comment:
Why is this noexcept?

Reply:
it only performs a possible deallocation

Reply:
"moving" bytes from the writable to the readable area merely adjusts a few pointers